pyrdp/player/LiveEventHandler.py

```diff
@@ -25,18 +28,43 @@ class LiveEventHandler(PlayerEventHandler, DirectoryObserver):
     file read events.
     """
 
-    def __init__(self, viewer: QRemoteDesktop, text: QTextEdit, log: LoggerAdapter, fileSystem: FileSystem, layer: PlayerLayer):
+    def __init__(self, viewer: QRemoteDesktop, text: QTextEdit, log: LoggerAdapter, fileSystem: FileSystem, layer: PlayerLayer, connectionRenameTab: Signal, tabInstance: LiveTab):
         super().__init__(viewer, text)
         self.log = log
         self.fileSystem = fileSystem
         self.layer = layer
         self.drives: Dict[int, Drive] = {}
         self.downloadFiles: Dict[str, BinaryIO] = {}
         self.downloadDialogs: Dict[str, FileDownloadDialog] = {}
+        self.connectionRenameTab = connectionRenameTab
+        self.tabInstance = tabInstance
 
         self.handlers[PlayerPDUType.DIRECTORY_LISTING_RESPONSE] = self.handleDirectoryListingResponse
         self.handlers[PlayerPDUType.FILE_DOWNLOAD_RESPONSE] = self.handleDownloadResponse
         self.handlers[PlayerPDUType.FILE_DOWNLOAD_COMPLETE] = self.handleDownloadComplete
+        self.handlers[PlayerPDUType.CLIENT_DATA] = self.onClientData
+        self.handlers[PlayerPDUType.CONNECTION_CLOSE] = self.onConnectionClose
+
+
+    def onClientData(self, pdu: PlayerPDU):
+        """
+        Message the LiveWindow to rename the tab to the hostname of the client
+        """
+
+        clientDataPDU = ClientConnectionParser().parse(pdu.payload)
+        clientName = clientDataPDU.coreData.clientName.strip("\x00")
```

Comment:
you might want to use `decodeUTF16LE()` from `helper.py` instead :)

Reply:
Parsing this string as UTF-16 gives garbage. Plus, I used that code from somewhere else in the code base.  This is already a parsed PDU, it has already been decoded as UTF-16LE

Reply:
Ahh, I didnt remember it could give trailing null bytes!